### ISSUE TITLE
Decimal Unmasked Value

### ISSUE THREAD

**Issue by bmw46033** [RobinHerbots/Inputmask#351]
It appears unmasked value for Decimal returns the comma (group separator) as part of the value.  Any way to turn that off?  

I want to show the comma during input but when I store the value in a database, I would like just the numbers.  I suppose I can replace comma with empty string, but I am implementing the mask in a generic input control.  I prefer to manipulate the behavior through an option.

thanks in advance,
Manash

**Comment by RobinHerbots** [comment]
@bmw46033 , 

I will do this in rel 2.7

**Comment by mtsunu** [comment]
is there a temporary work around for this?

**Comment by RobinHerbots** [comment]
Can't you store the value as decimal in the database?

Maybe you can parse the value to numeric with jquery.globalize

2013/11/21 Meigire Triassunu notifications@github.com

**Comment by mtsunu** [comment]
no.. because I'm using "." as group separator, so 1.000.000 will be stored as 1.00 in the database.

I'm using $.serialize to post the data via ajax. so I can't manually parse the data. so right now i do:

```
$(el).inputmask('remove'); $(el).removeData('_inputmask');
```

before calling $.serialize. and then re-attach the inputmask. I'm removing the _inputmask data manually because the inputmask didn't get removed immediately after calling remove
not sure if this is correct though..

also, i'm using jquery.validate plugin which doesn't provide before validate event. so i have to create custom method to validate the masked element.

**Comment by RobinHerbots** [comment]
You should set the groupseparator depending the used culture.  You can use
jquery.Globalize for that.

Something like

set data-inputmask attribute with "'alias': 'decimal'"

 //define uiculture
            $(":input").inputmask(undefined, { radixPoint:
Globalize.culture().numberFormat["."] });  //and offcourse also set the
separator

ifor jquery.validate you override the validators to also use
jquery.globalize

   $.extend($.validator.methods, {
   // http://docs.jquery.com/Plugins/Validation/Methods/number
        number: function (value, element) {
            var pvalue = Globalize.parseFloat(value);
            return this.optional(element) ||
/^-?(?:\d+|\d{1,3}(?:,\d{3})+)?(?:.\d+)?$/.test(pvalue);
        },

```
    // http://docs.jquery.com/Plugins/Validation/Methods/digits
    digits: function (value, element) {
        var pvalue = Globalize.parseInt(value);
        return this.optional(element) || /^\d+$/.test(value);
    },
```

and so on.

another possibility is to set autounmask to true and patch the unmasking
function to treat the decimals as separate, ... stripping the "."

I will have a relook at the issue later as I want to concentrate on the
dynamic masks and email mask.

2013/11/21 Meigire Triassunu notifications@github.com

**Comment by mtsunu** [comment]
I don't understand how jquery.globalize will help me.. I wan't to use "." as group separator, i don't care about the culture. and I can't change the database config.

looking at the source code, maybe I can do something with the isMask function?

**Comment by RobinHerbots** [comment]
Can you post some example to make it more clear.  (the exact mask you are using with which options

**Comment by mtsunu** [comment]
```
$('input[data-toggle="money-input"]').inputmask("decimal", { 
        radixPoint: ",", 
        autoGroup: true, 
        groupSeparator: ".", 
        groupSize: 3, 
        autoUnmask: true
    });
```

**Comment by RobinHerbots** [comment]
http://jsfiddle.net/Xfe9x/

**Comment by bmw46033** [comment]
$('#test').inputmask("decimal", { 
        radixPoint: ",", 
        autoGroup: true, 
        groupSeparator: ".", 
        groupSize: 3, 
        autoUnmask: true
    });

$('#btn').click(function(){
    var data = $("#test").inputmask('unmaskedvalue');

```
$('#res').val(data);
return false;
```

});

When I run this, the unmasked value still has the separators.  Ideally, would be nice to just get the raw number.

**Comment by mtsunu** [comment]
> Ideally, would be nice to just get the raw number.

yes this is what I want..

**Comment by iltoga** [comment]
I would like to give a little contribution to this amazing plugin that I found really useful.
I wrote a small patch (using SVN netbeans plugin) to be applied to the un-minified js (I use jquery.inputmask.bundle.js Version: 2.4.6).
This, after value has been  unmasked, will remove all occurrencies of the groupSeparator that has been defined during plugin initialization: 

``` javascript

Index: jquery.inputmask.bundle.js
--- jquery.inputmask.bundle.js Base (BASE)
+++ jquery.inputmask.bundle.js Locally Modified (Based On LOCAL)
@@ -106,7 +106,23 @@
                             masksets = $input.data('_inputmask')['masksets'];
                             activeMasksetIndex = $input.data('_inputmask')['activeMasksetIndex'];
                             opts = $input.data('_inputmask')['opts'];
-                            return maskScope(masksets, activeMasksetIndex).unmaskedvalue($input);
+                            /**
+                             * @note modified code to allow unmasking of group separator in numeric and decimal fields
+                             * @code
+                             * orginal code:
+                             * return maskScope(masksets, activeMasksetIndex).unmaskedvalue($input);
+                             * @endcode
+                             * STEF BEGIN
+                             */
+                            var unmaskedInput = maskScope(masksets, activeMasksetIndex).unmaskedvalue($input);
+                            if((opts.isNumeric || opts.alias === 'decimal') && opts.groupSeparator !==''){
+                                var re = new RegExp(opts.groupSeparator,"g");
+                                unmaskedInput = unmaskedInput.replace(re, "");
+                            }
+                            return unmaskedInput;
+                            /**
+                             * STEF END
+                             */
                         } else return $input.val();
                     case "remove":
                         return this.each(function () {

```

**Comment by RobinHerbots** [comment]
@iltoga,

Thank you for the contribution!

I'm going to add an onUnmask fn that will allow the above from within given options.  This will give more freedom of how the unmasking should behave.

**Comment by RobinHerbots** [comment]
@bmw46033 , @mtsunu, @iltoga,

Fixed in the current version

**Comment by t0n1zz** [comment]
hi sorry for opening this, but i still got the problem... i'm just downloaded your plugin a hour ago and using full bundle one... so? whats happening? i just need it to return a raw value to me not the input masked one

**Comment by RobinHerbots** [comment]
Trad the readme for the numeric alias.  Use autounmask with unmaskasanumber

**Comment by t0n1zz** [comment]
can you give me a example with html tags? because i try this and not working still when i type in there 10,000 only 10 that will be saved to database

```
<input data-inputmask=" 'alias' : 'numeric' , 'groupSeparator' : ',' , 'autoGroup' : true , 'autoUnmask' : true , 'unmaskAsNumber' : true" />
```

**Comment by RobinHerbots** [comment]
@t0n1zz , 

There seems to be a problem with the submit event binding. I will have a look to fix this.

**Comment by RobinHerbots** [comment]
@t0n1zz,

Grab the latest and use data-inputmask="'alias': 'numeric', 'groupSeparator': ',' , 'autoGroup': true, 'autoUnmask': true , 'removeMaskOnSubmit': true"
